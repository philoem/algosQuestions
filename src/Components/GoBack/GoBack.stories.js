import React from 'react'
import GoBack from './index'
import { withRouter } from 'storybook-addon-react-router-v6'

export default {
	component: GoBack,
	title: 'Components/GoBack',
	decorators: [withRouter],
	parameters: {
		reactRouter: {
			routePath: '/'
		}
	}
}

const Template = (args) => <GoBack {...args} />

export const Default = Template.bind({})
