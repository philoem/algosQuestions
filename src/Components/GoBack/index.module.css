.containerBtnGoBack {
  display: flex;
  justify-content: flex-start;
  padding-left: 1rem;
}

.btnGoBack {
  background-color: #61dafb;
  color: teal;
  padding-left: 1rem; /* 16px */
  padding-right: 1rem; /* 16px */
  padding: 0.5rem; /* 8px */
  font-weight: bold;
  height: 40px;
  border: 1px solid rgba(255, 255, 255, 0);
  border-radius: 8px;
  align-items: center;
}

.btnGoBack:hover {
  background-color: #2a265f;
  color: white;
}