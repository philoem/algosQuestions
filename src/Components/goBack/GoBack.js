import React from 'react'
import { useNavigate } from 'react-router-dom'
import styles from './GoBack.module.css'

function GoBack() {
	const navigate = useNavigate()
	const goTo = () => {
		navigate('..')
	}
	return (
		<div className={styles.containerBtnGoBack}>
			<button type="button" className={styles.btnGoBack} onClick={() => goTo()}>
				Back to home
			</button>
		</div>
	)
}

export default GoBack
