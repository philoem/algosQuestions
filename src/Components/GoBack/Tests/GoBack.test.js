//import { render, screen } from '@testing-library/react'
/**
 * @jest-environment jsdom
 */
import React from 'react'
import { render } from '@testing-library/react'
import { defineFeature, loadFeature } from 'jest-cucumber'
import GoBack from '../index'

const feature = loadFeature('src/Components/GoBack/Tests/GoBack.feature')

defineFeature(feature, (test) => {
	let container

	const givenGoBackButtonIsMounted = (given, comp) => {
		given('GoBack button is mounted', () => {
			container = render(comp).container
		})
	}

	test('GoBack button is displayed', ({ given, and }) => {
		givenGoBackButtonIsMounted(given, <GoBack />)
		and('Style should be applied', () => {
			expect(container).toMatchSnapshot()
		})
	})
})

// test('renders learn react link', () => {
// 	render(<GoBack />)
// 	const linkElement = screen.getByText(/Back to home/i)
// 	expect(linkElement).toBeInTheDocument()
// })
