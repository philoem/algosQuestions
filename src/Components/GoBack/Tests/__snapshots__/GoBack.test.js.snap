// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBack Button GoBack button is displayed 1`] = `
<div>
  <div
    class="containerBtnGoBack"
  >
    <button
      class="btnGoBack"
      type="button"
    >
      Back to home
    </button>
  </div>
</div>
`;
